trinity/sync/common/chain.py

```diff
@@ -184,7 +148,7 @@ class PeerHeaderSyncer(BaseService):
     def __init__(self,
                  chain: AsyncChain,
                  db: AsyncHeaderDB,
-                 peer: HeaderRequestingPeer,
+                 peer: BaseChainPeer,
```

Comment:
You might pull these type changes out to a small PR to drop a few things from the diff.